pkg/collector/corechecks/servicediscovery/envs/envs.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package envs provides target environment variables of interest.
+package envs
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/language"
+)
+
+// envKind separates environment variables by type, it helps to know when all target variables
```

Comment:
This optimization does not look correct. We can't know that just getting one `appEnv` is sufficient. Just to give one example the gunicorn parser falls back to `WSGI_APP` if there were problems in parsing `GUNICORN_CMD_ARGS`, so it's wrong to stop after getting `GUNICORN_CMD_ARGS`. I'd suggest you remove this optimization and just gather all the environment variables in the list.

Reply:
I've removed search optimization in last commit